### PR TITLE
Use etcd v3.4 for Kube 1.17+

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

* Upgrades etcd to 3.4 for Kube 1.17+ clusters
* Changes the backup cronjob to also work with the new etcd version

I have manually checked that:
* Upgrading etcd works
* The backup job works for both etcd 3.4 and 3.3
* The defragger job works for both etcd 3.4 and 3.3

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Etcd was upgraded to 3.4 for 1.17+ clusters
```

/CC @shahbazn

### CODE REVIEW

File: api/pkg/controller/seed-controller-manager/backup/backup_controller.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ const (
 	// will write the backup to
 	SharedVolumeName = "etcd-backup"
 	// DefaultBackupContainerImage holds the default Image used for creating the etcd backups
-	DefaultBackupContainerImage = "gcr.io/etcd-development/etcd:" + etcd.ImageTag
+	DefaultBackupContainerImage = "gcr.io/etcd-development/etcd:"
```
Comment (xrstf) [inline api/pkg/controller/seed-controller-manager/backup/backup_controller.go]: This should break the image-loader. I also don't think having the ":" at the end here is a good idea.

Comment (xrstf) [reply api/pkg/controller/seed-controller-manager/backup/backup_controller.go]: What happens if someone runs the seed-controller with a custom backup container image that contains a version already? Is this a BC break and should be documented?

Comment (alvaroaleman) [reply api/pkg/controller/seed-controller-manager/backup/backup_controller.go]: Changed it to check for a `:` before adding a tag. Also removed this from the imageloader, as its the same image used by the etcd statefulset (par definition) and that is already being pulled in.

**Pull request merged**